library/vulnerabilities/path-traversal/unsafePathStart.ts

```diff
@@ -22,12 +24,25 @@ const linuxRootFolders = [
 const dangerousPathStarts = [...linuxRootFolders, "c:/", "c:\\"];
 
 export function startsWithUnsafePath(filePath: string, userInput: string) {
-  const lowerCasePath = filePath.toLowerCase();
-  const lowerCaseUserInput = userInput.toLowerCase();
+  // Check if path is relative (not absolute or drive letter path)
+  // Required because resolve will build absolute paths from relative paths
+  if (!isAbsolute(filePath)) {
+    return false;
+  }
+
+  let origResolve = resolve;
+  // @ts-expect-error __wrapped is not typed
+  if (resolve.__wrapped) {
+    // @ts-expect-error Not type safe
+    origResolve = resolve.__original;
+  }
+
+  const normalizedPath = origResolve(filePath).toLowerCase();
+  const normalizedUserInput = origResolve(userInput).toLowerCase();
```

Comment:
is it possible that the resolve might throw if we pass weird user strings to it?

Reply:
Also tested `resolve(...)` it with all kinds of strings, including emojis 😄, it did not throw.